web/auth.go

```diff
@@ -58,6 +71,83 @@ func SignupPageHandler(w http.ResponseWriter, r *http.Request) {
 	tmp.Execute(w, data)
 }
 
+func LoginHandler(w http.ResponseWriter, r *http.Request) {
+	// check if POST data are set and validate them
+}
+
+// SignupHandler handles the signup process for in app signup
+func SignupHandler(w http.ResponseWriter, r *http.Request) {
+	// I feel SignupPageHandler() and this function should share the same struct for better memory conservation
+	data := struct {
+		FBAuthURL     string
+		GoogleAuthURL string
+		SignupError   string
+	}{}
+
+	// This is set here so that when there are any errors from the signup process,
+	// the link will be passed to the template alongside the errors.
+	data.FBAuthURL = FBOauthConf.AuthCodeURL("state", oauth2.AccessTypeOffline)
+
+	r.ParseForm()
+
+	fullName := r.FormValue("name")
+	email := r.FormValue("email")
+	passwd := r.FormValue("passwd")
+
+	// Check if the Post data not empty and validate them.
+	if fullName != "" && email != "" && passwd != "" {
```

Comment:
revert this line. The bulk of the function should not be in an if clause. Instead run 

`
if fullName==""||email==""||passwd==""{
   data.SignupError = "Please Fill out all required Fields"
 	tmp := GetTemplates().Lookup("signin_signup.html")
 	tmp.Execute(w, data)
        return
}
`
then remove the current content of the function into the default scope.

Learn the habit of keeping the larger code area in the root scope.  Else you'd quickly have unmaintainable code